src/app/wallet/wallet/shared/transaction.service.ts

```diff
@@ -83,45 +94,51 @@ export class TransactionService {
   rpc_update() {
 
     const options = {
-      'count': +this.MAX_TXS_PER_PAGE,
-      'skip': this.currentPage * this.MAX_TXS_PER_PAGE
+      'count':            +this.MAX_TXS_PER_PAGE,
+      'skip':             +this.MAX_TXS_PER_PAGE * this.currentPage,
+      'include_watchonly': this.watchonly !== undefined ? this.watchonly : undefined,
+      'category':          this.category  !== undefined ? this.category  : undefined,
+      'search':            this.search    !== undefined ? this.search    : undefined,
+      'type':              this.type      !== undefined ? this.type      : undefined,
+      'sort':              this.sort      !== undefined ? this.sort      : undefined
     };
+
     this.rpc.call('filtertransactions', [options])
-    .subscribe(
-      (txResponse: Array<Object>) => {
-        // The callback will send over an array of JSON transaction objects.
-        this.log.d(`rpc_loadTransactions_success, supposedly tx per page: ${this.MAX_TXS_PER_PAGE}`);
-        this.log.d(`rpc_loadTransactions_success, real tx per page: ${txResponse.length}`);
-
-        if (txResponse.length !== this.MAX_TXS_PER_PAGE) {
-          this.log.er(`rpc_loadTransactions_success, TRANSACTION COUNTS DO NOT MATCH (maybe last page?)`);
-        }
-
-          this.deleteTransactions();
-          txResponse.forEach((tx) => {
-            this.addTransaction(tx);
-          });
-
-        this.loading = false;
-        this.log.d(`rpc_update, txs array: ${this.txs.length}`);
+    .subscribe((txResponse: Array<Object>) => {
+      // The callback will send over an array of JSON transaction objects.
+      this.log.d(`rpc_loadTransactions_success, supposedly tx per page: ${this.MAX_TXS_PER_PAGE}`);
+      this.log.d(`rpc_loadTransactions_success, real tx per page: ${txResponse.length}`);
+
+      if (txResponse.length !== this.MAX_TXS_PER_PAGE) {
+        this.log.er(`rpc_loadTransactions_success, TRANSACTION COUNTS DO NOT MATCH (maybe last page?)`);
+      }
+
+      this.deleteTransactions();
+      txResponse.forEach((tx) => {
+        this.addTransaction(tx);
```

Comment:
maybe the paginator should be updated here ?

Reply:
tx count is not updated after API call so could you please confirm me how can I get updated tx count after filter applied?

Reply:
Please update tx record state is not updated after getting filter record. please update state then pagination will be work fine.

Reply:
@anandsinghparihar `txResponse.length` is the new total

Reply:
I guess `txResponse.length`  is returns total item per page?